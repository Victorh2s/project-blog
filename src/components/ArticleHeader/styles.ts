import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`

  text-align: center;
  justify-content: center;
  margin-bottom: ${theme.spacings.xlarge};
  border-bottom: 0.1rem solid ${theme.colors.darkeyGray};

  ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }

  @media ${theme.media.lteMedium} {
      ${HeadingStyles}{
      font-size: ${theme.font.sizes.large};
    }
  }

`}
`;

export const Text = styled.p`
  text-align: center;
  justify-content: center;
`;

export const Img = styled.img`
  max-width: 100%;
  display: block;
`;
