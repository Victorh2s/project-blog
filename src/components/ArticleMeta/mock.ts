import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-06-10T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'victor-henrique',
    displayName: 'Victor H',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'Tech',
      slug: 'tech',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
