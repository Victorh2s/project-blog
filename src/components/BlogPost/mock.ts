import { BlogPostProps } from '.';
import { data } from '../api/dados.json';

const { title, excerpt, content, cover, author, categories, createdAt } =
  data.posts[0];

export default {
  title,
  excerpt,
  content,
  cover,
  author,
  categories,
  createdAt,
} as BlogPostProps;
