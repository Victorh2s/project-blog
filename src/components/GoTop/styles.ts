import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    color: ${theme.colors.primary};
    border: 0.1rem solid;
    border-radius: 50rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
