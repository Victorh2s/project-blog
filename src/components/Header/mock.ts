import { HeaderProps } from '.';

export default {
  BlogName: 'Usu√°rio teste',
  BlogDescription:
    'Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut.',
  logo: 'assets/images/logo.png',
  showText: true,
} as HeaderProps;
