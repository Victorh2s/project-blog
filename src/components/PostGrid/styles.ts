import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  width: 100%;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
  max-width: 120rem;
  max-height: 300rem;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
  gap: ${theme.spacings.large};
  padding: ${theme.spacings.large};
  text-overflow: ellipsis;

  @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
${({ theme }) => css`

    padding: ${theme.spacings.large};
    text-align: center;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.secondary};

`}
`;
