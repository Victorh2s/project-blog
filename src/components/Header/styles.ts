import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as LogoLinkstyles } from '../LogoLink/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: 96rem;
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    ${LogoLinkstyles}{
      >img{
        border: 0.3rem solid;
        border-color:  rgb(74 222 128);
      }
    }

    @media ${theme.media.lteMedium}{
      display:flex;
      flex-flow: column wrap;

      & > ${HeadingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }

    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.lteMedium}{
      margin-left: 0;
    }
  `}
`;
