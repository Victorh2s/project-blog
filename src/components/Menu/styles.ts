import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  Visible: boolean;
};

const ContainerChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? 0 : '-35rem'};
  overflow-y: ${Visible ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '0' : '-35rem'};
  }
`;

export const Container = styled.div<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${ContainerChanger(Visible, theme)}
  `}
`;

export const Nav = styled.nav`
    margin: 0 auto;
    width: 100%;
`;

export const MenuLinks = styled.div`

   margin-bottom: 5rem;

`;

export const Logo = styled.div`

    ${HeadingStyles} {

      img{
        margin: 0 auto;
        margin-bottom: 3rem;
      }
    }

`;

const ButtonChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? '26rem,' : '1rem'};
  color: ${Visible ? theme.colors.secondary : theme.colors.primary};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '26rem' : '-0.5rem'};
  }
`;

export const Button = styled.a<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    left: 26rem;
    color: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    transition: all 300ms ease-in-out;
    ${ButtonChanger(Visible, theme)}
  `}
`;
