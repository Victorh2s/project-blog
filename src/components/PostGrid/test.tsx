import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostGrid, PostGridProps } from '.';
import mock from './mock';

const props: PostGridProps = mock;

describe('<PostGrid />', () => {
  it('should not render NotFound', () => {
    renderTheme(<PostGrid {...props} />);

    expect(screen.queryByText(/Nenhum post/i)).not.toBeInTheDocument();
  });

  it('should render NotFound', () => {
    renderTheme(<PostGrid {...props} posts={[]} />);

    expect(screen.queryByText(/Nenhum post/i)).toBeInTheDocument();
  });

  it('should render posts', () => {
    const { container } = renderTheme(<PostGrid {...props} />);

    expect(screen.getAllByRole('heading')).toHaveLength(3);
    expect(screen.getAllByRole('img')).toHaveLength(3);
    expect(screen.queryByText(/Nenhum post/i)).not.toBeInTheDocument();

    expect(container.querySelectorAll('p')).toHaveLength(3);
  });

  it('should render snapshot', () => {
    const { container } = renderTheme(<PostGrid {...props} />);

    expect(container).toMatchSnapshot();
  });
});
