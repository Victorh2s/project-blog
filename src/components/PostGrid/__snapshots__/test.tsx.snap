// Vitest Snapshot v1

exports[`<PostGrid /> > should render snapshot 1`] = `
.c5 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: none;
}

.c2 .c4 {
  margin: 0;
  margin-top: 1.6rem;
  margin-bottom: 1.6rem;
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2:hover a {
  color: #dc143c;
}

.c2:hover img {
  opacity: 0.8;
}

.c3 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c0 {
  width: 100%;
}

.c1 {
  max-width: 120rem;
  max-height: 300rem;
  box-shadow: rgba(0,0,0,0.35) 0px 5px 15px;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
  text-overflow: ellipsis;
}

@media (max-width:360px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="/post/vis-nobis-veritus-maluisset-id-nonumy-adversarium"
        >
          <img
            alt="Vis nobis veritus maluisset id, nonumy adversarium"
            class="c3"
            src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/vis-nobis-veritus-maluisset-id-nonumy-adversarium"
          >
            Vis nobis veritus maluisset id, nonumy adversarium
          </a>
        </h2>
        <p
          class=""
        >
          Fugit aliquip erroribus eu eam, an mel vidit doctus. Vis nobis veritus maluisset id, nonumy adversarium neglegentur est ut. Tation persius pri eu, modo minim fastidii mei no. Ad tota pertinax definiebas eam.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/sea-array-destructuring-dicat-fabulas-consetetur"
        >
          <img
            alt="Sea Array Destructuring dicat fabulas consetetur "
            class="c3"
            src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614699042/Origin_of_nature_by_Julian_Tomasini_dcfdd17cbe.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/sea-array-destructuring-dicat-fabulas-consetetur"
          >
            Sea Array Destructuring dicat fabulas consetetur 
          </a>
        </h2>
        <p
          class=""
        >
          Quod oblique mei ne, cetero delicatissimi sea cu, cu est harum dicit verterem. Tota forensibus ut usu, luptatum dignissim persecuti vel ex. Mea nostrud instructior cu, et referrentur theophrastus eum. Pro in iuvaret senserit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/quod-oblique-mei-ne-cetero-delicatissimi-sea"
        >
          <img
            alt="Quod oblique mei ne, cetero delicatissimi sea"
            class="c3"
            src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614699042/Origin_of_nature_by_Julian_Tomasini_dcfdd17cbe.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/quod-oblique-mei-ne-cetero-delicatissimi-sea"
          >
            Quod oblique mei ne, cetero delicatissimi sea
          </a>
        </h2>
        <p
          class=""
        >
          Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut. 
        </p>
      </div>
    </div>
  </div>
</div>
`;
